### DISCUSSION TITLE
Dynamic SEO for 100,000 profile pages - use react-static?

### DISCUSSION THREAD

**Discussion by augustmuir** [react-static/react-static#1632] [Q&A]
I used react-static for my last website and it worked great, started to get organic traffic within a couple days after switching from create-react-app, the website was simple with less than 50 pages.

My new project has the requirements to be SEO friendly for `n` user pages - including dynamic titles, descriptions, content, and URLs. I need it to work kind of like Instagram or TikTok's SEO, example:

If you google `champagnepapi instagram` it will return:
URL: `https://www.instagram.com/champagnepapi/`
Google Title: `@champagnepapi • Instagram photos and videos`
Google Description: `93.4m Followers, 2599 Following, 5115 Posts - See Instagram photos and videos from @champagnepapi.`

So my question is - would you recommend react-static for this requirement?

Edit: I researched and decider SSR and Next.js would be optimal for this, feel free to delete this discussion.

**Comment by SleeplessByte** [comment]
In the past I did a react-static site for 250k pages. I cannot recommend it because of the long build / upload time. Next.js is _really_ good at this, especially with their ISR.
